### PR TITLE
[Maintenance] Add conflict with jms/serializer-bundle 4.1.0

### PR DESCRIPTION
**Pull request by coldic3**
| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | 1.12                  |
| Bug fix?        | yes                                                       |
| New feature?    | no                                                       |
| BC breaks?      | no                                                       |
| Deprecations?   | no |
| Related tickets |                       |
| License         | MIT                                                          |

See https://github.com/Sylius/Sylius/runs/8306007639?check_suite_focus=true and https://github.com/schmittjoh/JMSSerializerBundle/releases/tag/4.1.0

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -178,7 +178,7 @@
     },
     "conflict": {
         "doctrine/doctrine-bundle": "2.3.0",
-        "jms/serializer-bundle": "3.9.0",
+        "jms/serializer-bundle": "3.9.0 || 4.1.0",
```
Comment (loic425) [inline composer.json]: I hope it'll be fixed in 4.1.1...

**Review by loic425 — APPROVED** [review]

**Review by NoResponseMate — APPROVED** [review]

**Review by TheMilek — APPROVED** [review]

File: composer.json (hunk 1 above)
Comment (GSadee) [reply composer.json]: If you are adding a new conflict, add also a note to `CONFLICTS.md` file with short explanation

Comment (lchrusciel) [reply composer.json]: What is more, our current requirement makes the left part of this conflict unreachable (^4.0). Lets remove it as well
image

Comment (coldic3) [reply composer.json]: Thanks! Done 🙂

File: CONFLICTS.md [hunk 2]
```diff
@@ -24,10 +24,10 @@ references related issues.
...
 
- - `jms/serializer-bundle:3.9`:
+ - `jms/serializer-bundle:4.1.0`:
 
-   This version automatically registered DocBlockDriver, which is always turned on, while docblocks used in our code are not usable with it. Sample error:
-   `Can't use incorrect type object for collection in Doctrine\ORM\PersistentCollection:owner`
+   This version contains service with a wrong constructor arguments:
+   `Invalid definition for service ".container.private.profiler": argument 4 of "JMS\SerializerBundle\Debug\DataCollector::__construct()" accepts "JMS\SerializerBundle\Debug\TraceableDriver", "JMS\SerializerBundle\Debug\TraceableMetadataFactory" passed.`
 
    References: https://github.com/schmittjoh/JMSSerializerBundle/issues/844
```
Comment (GSadee) [inline CONFLICTS.md:32]: I guess that this reference should be removed or changed to the current issue

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thank you, Kevin! :1st_place_medal:

**Pull request merged**